### PR TITLE
feat: filter out inactive users from list and add option to see all users

### PR DESCRIPTION
**Pull request by AmanuelAaron**
## Description

We want to be able to list out active users only in the CLI when running `det u`. We also want the option to list all active and inactive users with an additional `--all` flag

## Test Plan

Run `det u` and `det u --a` to verify if active users and all users are listed respectively

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-9145

## End-User Docs

#8479

### CODE REVIEW

File: docs/release-notes/list-active-users.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+:orphan:
+
+**New Features**
+
+-  CLI: ``det users ls`` now only shows active users. New flag ``--all`` added which will show all users.
```
Comment (tara-det-ai) [inline docs/release-notes/list-active-users.rst:5]: :orphan:

**New Features**

-  CLI: Modify ``det users ls`` to show only active users. Add a new flag ``--all`` to show all users.

**Review by tara-det-ai — COMMENTED** [review]
suggested edits

File: harness/determined/experimental/client.py [hunk 2]
```diff
@@ -289,10 +289,13 @@ def logout() -> None:
 
 
 @_require_singleton
-def list_users() -> List[User]:
+def list_users(active_only: bool) -> List[User]:
```
Comment (ioga) [inline harness/determined/experimental/client.py:292]: this should be the same exact interface as the class method (but singleton)
```suggestion
def list_users(active: Optional[bool] = None) -> List[User]:
```

File: harness/determined/experimental/client.py [hunk 3]
```diff
@@ -289,10 +289,13 @@ def logout() -> None:
...
     """Get a list of all Users."""
     assert _determined is not None
-    return _determined.list_users()
+    if active_only:
+        return _determined.list_users(active=True)
+    else:
+        return _determined.list_users()
```
Comment (ioga) [inline harness/determined/experimental/client.py:298]:
```suggestion
    return _determined.list_users(active=active)
```

File: docs/release-notes/list-active-users.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**New Features**
+
+-  CLI: Modify ``det users`` ls to show only active users. Add a new flag ``--all`` to show all
+   users.
```
Comment (ioga) [inline docs/release-notes/list-active-users.rst:6]:
```suggestion
-  CLI: By default, ``det user ls`` will now only display active users. ``det user ls --all`` can be used to display all users.
```

File: harness/determined/cli/user.py [hunk 5]
```diff
@@ -45,9 +45,9 @@
 def list_users(args: Namespace) -> None:
     resp = bindings.get_GetMaster(setup_session(args))
     if resp.to_json().get("rbacEnabled"):
-        render.render_objects(FullUserNoAdmin, client.list_users())
+        render.render_objects(FullUserNoAdmin, client.list_users(active_only=not args.all))
```
Comment (ioga) [inline harness/determined/cli/user.py:48]:
```
        users_list = client.list_users(active=None if args.all else True)
        renderer = FullUser
        if resp.to_json().get("rbacEnabled"):
            renderer = FullUserNoAdmin
        render.render_objects(renderer, users_list)
```

File: harness/determined/experimental/client.py (hunk 3 above)
Comment (stoksc) [reply harness/determined/experimental/client.py:298]: active=False != all users

Comment (stoksc) [reply harness/determined/experimental/client.py:298]: oh i see what you mean, this plus the comment above. gotcha.

Comment (rb-determined-ai) [reply harness/determined/experimental/client.py:298]: writing a unit test for it is preferred.  There's not a lot of examples of that just yet, unfortunately.

File: harness/determined/cli/user.py (hunk 5 above)
Comment (AmanuelAaron) [reply harness/determined/cli/user.py]: This implementation fails the linter as renderer is being set as two different types (FullUser & FullUserNoAdmin)

Comment (ioga) [reply harness/determined/cli/user.py]: `renderer = FullUser  # type: Any`?

Comment (AmanuelAaron) [reply harness/determined/cli/user.py]: Thanks. Updated!

File: docs/release-notes/list-active-users.rst [hunk 6]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**New Features**
+
+-  CLI: Modify ``det user ls`` ls to show only active users. Add a new flag ``--all`` to show all
+   users.
```
Comment (tara-det-ai) [inline docs/release-notes/list-active-users.rst:6]: The first part sounds like an improvement or a breaking change. The second part sounds like a new feature.

Comment (tara-det-ai) [reply docs/release-notes/list-active-users.rst:6]: i'll update the documentation via PR #8479

**Review by ioga — APPROVED** [review]

**Review by stoksc — APPROVED** [review]
i don't think you're waiting on me, but just incase, lg

**Review by tara-det-ai — APPROVED** [review]
LGTM, one comment added

**Pull request merged**